libs/platform/src/lib/components/thumbnail/thumbnail-image/thumbnail-image.component.ts

```diff
@@ -1,39 +1,82 @@
-import { Component, Input, Output, EventEmitter, ViewEncapsulation, OnChanges, SimpleChanges } from '@angular/core';
+import { Component, Input, Output, EventEmitter, ViewEncapsulation, OnChanges, SimpleChanges, OnInit, ChangeDetectionStrategy } from '@angular/core';
+import { DialogService, RtlService } from '@fundamental-ngx/core';
 import { Media } from '../thumbnail.component';
-
+import { ThumbnailDetailsComponent } from '../thumbnail-details/thumbnail-details.component';
 @Component({
     selector: 'fdp-thumbnail-image',
     templateUrl: './thumbnail-image.component.html',
-    styleUrls: ['./thumbnail-image.component.scss'],
-    encapsulation: ViewEncapsulation.None,
+    styleUrls: ['./thumbnail-image.component.scss']
+
 })
-export class ThumbnailImageComponent implements OnChanges {
+export class ThumbnailImageComponent implements OnChanges, OnInit {
 
+    /** media list obejct contains group of Media object.*/
     @Input()
     mediaList: Media[];
 
+    /** Boolean to set the orientaiton of thumbnail images.*/
     @Input()
     isHorizontal = false;
 
+    /**Maximum limit for the thumbnail images to display */
+    @Input()
+    maxImages = 5;
+
+    /** Output event for thumbnail image click */
     @Output()
     thumbnailClicked: EventEmitter<Media> = new EventEmitter();
 
+    /** @hidden */
+    constructor(private _dialogService: DialogService, private _rtlService: RtlService) { }
+
+    /** @hidden */
+    ngOnInit(): void {
+        if (this.mediaList.length > this.maxImages) {
+            this.mediaList[this.maxImages - 1].overlayRequired = true;
+        }
```

Comment:
please move it to the method and reuse it in both places

Reply:
had created private method.